src/main/java/com/gluonhq/substrate/util/FileDeps.java

```diff
@@ -201,6 +205,43 @@ public static boolean setupDependencies(ProjectConfiguration configuration) thro
         return true;
     }
 
+    /**
+     * Returns the path to the llc compiler that is working for the provided configuration.
+     * The <code>configuration</code> object must have its host triplet set correctly.
+     * If the llc compiler is found in the file cache, it will be returned. Otherwise, it will
+     * be downloaded and stored in the cache. After calling this method, it is guaranteed that an
+     * llc compiler is in the specified path.
+     * <p>
+     *     There might be different versions for the llc compiler, but this is handled inside
+     *     Substrate. If the developer wants a specific flavour of llc, it is recommended to
+     *     use <code>configuration.setLlcPath()</code> which takes precedence over calling this method.
+     * </p>
+     * @param configuration
+     * @return the path to a working llc compiler.
+     * @throws IOException in case the required directories can't be created or navigated into.
+     */
+    public static Path getLlcPath(ProjectConfiguration configuration) throws IOException {
+        Path llcRootPath = Constants.USER_SUBSTRATE_PATH.resolve("llc");
+        String archos = configuration.getHostTriplet().getArchOs();
+        Path archosPath = llcRootPath.resolve(archos).resolve(Constants.LLC_VERSION);
+        if (!archosPath.toFile().exists()) {
+            archosPath.toFile().mkdirs();
+        }
+        String llcname = "llc-"+archos+"-"+Constants.LLC_VERSION;
+        Path llcPath = archosPath.resolve(llcname);
+        if (llcPath.toFile().exists()) {
+            return llcPath;
+        }
+        // we don't have the required llc. Download it and store it in llcPath.
+        URL url = new URL(LLC_URL+llcname);
+        ReadableByteChannel readableByteChannel = Channels.newChannel(url.openStream());
```

Comment:
move to `try` with resources?

Reply:
I'm worried that will hide exceptions. I don't want to catch anything here. At this point, we don't care about filestreams not being closed (hence no final), as any exception here should be consider as fatal. try-with-resources is ok in case of non-fatal exceptions, but if anything goes wrong here, we don't even consider recovering.

Reply:
We can always add `try (...) { ... } catch (Exception ex)  { // handle exception }`, can't we?

Reply:
But what exception do we want to catch here?

Reply:
As is now, you are performing the download without any try-catch, only throwing `IOException`, should this happen.

I guess you can do something similar with:
```
try (...) { ... 
} catch (Exception ex) { 
    throw new IOException("Error downloading llc: " + e.getMessage()); 
}
```

Reply:
But since I don't catch any exceptions, it means they will propagate anyhow.